ui/app/templates/storage/index.hbs

```diff
@@ -0,0 +1,252 @@
+            @currentPage={{this.csiPage}}
+            @pageSizes={{this.pageSizes}}
+            @currentPageSize={{this.userSettings.pageSize}}
+            @onPageChange={{action this.handlePageChange "csi"}}
+            @onPageSizeChange={{action (mut this.userSettings.pageSize)}}
+          />
+        {{/if}}
+      {{else}}
+        <div class="empty-message" data-test-empty-csi-volumes-list-headline>
+          {{#if this.csiFilter}}
+            <p>No CSI volumes match your search for "{{this.csiFilter}}"</p>
+            <Hds::Button @text="Clear search" @color="secondary" {{on "click" (queue (action (mut this.csiFilter) "") (action this.handlePageChange "csi" 1))}} />
+          {{else}}
+            <p>No CSI Volumes found</p>
+          {{/if}}
+        </div>
+      {{/if}}
+    </Hds::Card::Container>
+
+    <Hds::Card::Container @level="base" @hasBorder={{false}} class="storage-index-table-card">
+      <header aria-label="Dynamic Host Volumes">
+        <h3>Dynamic Host Volumes</h3>
+        <p class="intro">
+          Storage provisioned via plugin scripts on a particular client, modifiable without requiring client restart.
+          <Hds::Link::Inline @href="https://developer.hashicorp.com/nomad/docs/other-specifications/volume/host" @icon="docs-link" @iconPosition="trailing">Read more</Hds::Link::Inline>
+        </p>
+        <div class="search">
+          <Hds::Form::TextInput::Field
+            data-test-dynamic-host-volumes-search
+            @type="search"
+            @value={{this.dhvFilter}}
+            placeholder="Search Dynamic Host Volumes"
+            {{on "input" (action this.applyFilter "dhv")}}
+          />
+        </div>
+      </header>
+      {{#if this.sortedDynamicHostVolumes.length}}
+        <Hds::Table @caption="Dynamic Host Volumes"
+          @model={{this.sortedDynamicHostVolumes}}
+          @columns={{this.dhvColumns}}
+          @sortBy={{this.dhvSortProperty}}
+          @sortOrder={{if this.dhvSortDescending "desc" "asc"}}
+          @onSort={{action this.handleSort "dhv"}}
+        >
+          <:body as |B|>
+            <B.Tr data-test-dhv-row
+              {{keyboard-shortcut
+                enumerated=true
+                action=(action this.openDHV B.data)
+              }}
+            >
+              <B.Td>
+                <LinkTo data-test-dhv-name={{B.data.name}} @route="storage.volumes.dynamic-host-volume"
+                  @model={{B.data.idWithNamespace}}>
+                    {{B.data.plainId}}
+                  </LinkTo>
+              </B.Td>
+              <B.Td>
+                {{B.data.name}}
+              </B.Td>
+              {{#if this.system.shouldShowNamespaces}}
+                <B.Td>{{B.data.namespace}}</B.Td>
+              {{/if}}
+              <B.Td>
+                <LinkTo @route="clients.client" @model={{B.data.node.id}}>
+                  {{B.data.node.name}}
+                </LinkTo>
```

Comment:
This may be a Mirage quirk, but if I visit the storage index page this value is empty. If I click into the detail page and then hit the back button, it appears.

Reply:
Yeah, I think this is just mirage. In order to get node .name, we fetch any clients that a DHV belongs to (immediately after rendering, so there can be a brief delay on the page).

Alternatives to this IMO are both less desirable:
- delay loading all page data until node data is loaded
- only show the node ID (which comes back with the DHV) instead of its name

I'll see what Mirage is up to to cause this, I'm able to reproduce.